Source/DiabloUI/credits.cpp

```diff
@@ -1,23 +1,38 @@
 #include <algorithm>
+#include <cstddef>
 #include <cstdint>
-#include <memory>
+#include <optional>
+#include <string>
+#include <string_view>
 #include <vector>
 
+#include <SDL_events.h>

```

Comment:
It's a bit unusual to include individual SDL headers rather than all of `#include <SDL.h>`. I wonder if it'll make any difference to SDL3 migration, where we'd need to do `#include <SDL3/SDL.h>`.

I don't havfe a strong opinion on this but `IgnoreHeaders` could be used here https://clang.llvm.org/extra/clang-tidy/checks/misc/include-cleaner.html

Reply:
This is up to your personal taste.
SDL supports inclusion of each header separately, but advises users to only include SDL.h for simplicity.

Keeping track of what SDL headers to include and keeping them updated is an extra maintenance burden,
but perhaps doable with tooling.

SDL is also a c-only library, so you don't have to worry about fancy SFINAE c++ template programming causing a massive compile time increase.

Reply:
In that case I think we should go with IgnoreHeaders, I'll update the PR to take this in to account.